### PR TITLE
Add support for name:pronunciation tag

### PR DESCRIPTION
**Pull request by danpaz**
# Issue

Adds support for the [`name:pronunciation`](https://wiki.openstreetmap.org/wiki/Key:name:pronunciation) OSM tag using the TaggedName structure introduced in #2608 . According to taginfo approximately [2,770](https://taginfo.openstreetmap.org/keys/?key=name%3Apronunciation) features include this tag. 

Closes https://github.com/valhalla/valhalla/issues/1148

/cc @1ec5

## Tasklist

 - [X] Add tests
 - [ ] Add #fixes with the issue number that this PR addresses
 - [ ] Generally use squash merge to rebase and clean comments before merging
 - [ ] Update the [changelog](CHANGELOG.md)

## Requirements / Relations

 Link any requirements here. Other pull requests this PR is based on?

### CODE REVIEW

File: src/mjolnir/osmway.cc [hunk 1]
```diff
@@ -182,6 +182,13 @@ std::vector<std::string> OSMWay::GetTaggedNames(const UniqueNames& name_offset_m
       names.emplace_back(std::to_string(static_cast<uint8_t>(TaggedName::kTunnel)) + t);
     }
   }
+  if (name_pronunciation_index_ != 0) {
+    // name:pronunciation names
+    tokens = GetTagTokens(name_offset_map.name(name_pronunciation_index_));
+    for (const auto& t : tokens) {
+      names.emplace_back(std::to_string(static_cast<uint8_t>(TaggedName::kPronunciation)) + t);
```
Comment (1ec5) [inline src/mjolnir/osmway.cc:189]: Does Valhalla still use tags like `alt_name`, `official_name`, and `name:en` in guidance instructions, as claimed in #2254? If so, [other `:pronunciation` tags](https://taginfo.openstreetmap.org/search?q=pronunciation) may be relevant, though `name:pronunciation` is the most prominent of them.

**Review by 1ec5 — COMMENTED** [review]
Glad to see pronunciation overrides make it in here!

**Review by kevinkreiser — APPROVED** [review]

File: src/mjolnir/osmway.cc (hunk 1 above)
Comment (kevinkreiser) [reply src/mjolnir/osmway.cc:189]: `alt_name` is a conditional config parameter when building tiles now so you can turn it off. `official_name` is still added as a name regardless of configuration in addition to `name` and also `name:en`. you can check this function for more details: https://github.com/valhalla/valhalla/blob/6add48fa3e283c159ab03f1bc5ed5b4722acc7a2/src/mjolnir/osmway.cc#L99

in the future we'll be able to store language specific tagging (now that we have tagged values support) so we should be able to "undo" the `name:en` fallback to be a bit more graceful

**Review by gknisely — DISMISSED** [review]

File: proto/tripcommon.proto [hunk 2]
```diff
@@ -150,6 +150,7 @@ message TaggedName {
   enum Type {
     kTunnel = 1;
     kBridge = 2;
+    kPronunciation = 3;
```
Comment (dgearhart) [inline proto/tripcommon.proto:153]: if this is for street name pronunciation - then should it be stored with the street name it is associated with?

**Review by dgearhart — CHANGES_REQUESTED** [review]
We should discuss - the pronunciation needs to be associated with its base meaning - street name, destination, etc

**Comment by dgearhart** [comment]
This PR has been superseded with https://github.com/valhalla/valhalla/pull/3132

**Pull request closed without merging**